src/main/java/uk/gov/hmcts/darts/common/entity/EventEntity.java

```diff
@@ -87,4 +87,9 @@ public class EventEntity extends CreatedModifiedBaseEntity {
     public void addHearing(HearingEntity hearingEntity) {
         hearingEntities.add(hearingEntity);
     }
-}
+
+    public HearingEntity associatedHearing() {
+        // Events can only be associated with one hearing at this stage
+        return this.hearingEntities.getFirst();
```

Comment:
I think we should add some logic here something like if(this.getHearingEntities().size() > 1) {
 throw new UnsupportedOperation(...);
}

or something just so if we do associate multiple hearings in the future this scenario becomes obvious

Reply:
We probably need a wider discussion on this as because the data model supports Many-To-Many relationships we should account for these and throw relevant errors where we are getting multiple items returned where we are only expecting a single item.

This just means if we ever get bad data on the system or we add support for multiple hearing / event associations we don't have code that hides failures

Reply:
Happy to throw an Exception if this is ok with Tech Leads. My understanding is that we wanted to handle this gracefully, also looking at other places. @hemantasharma1129 what do you think?

Reply:
For additional context we do also have logic to chuck an exception in this instance in other places>
So the question is do we want to be altered when data gets into an invalid state or do we want to gracefully consume it

<img width="571" alt="image" src="https://github.com/user-attachments/assets/1f0caffc-3c7d-4fac-a219-d74e9c106fba">

Reply:
I assume the list will never be empty. In such cases a [NoSuchElementException](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/NoSuchElementException.html) would be thrown

Reply:
We could protect against all the aforementioned scenarios by adding a check to the EventIdValidator. Better to gracefully handle these issues ourselves rather than throwing exceptions ourselves that will result in a 5xx response.

Reply:
Also maybe if this exception was to be thrown it should be a darts exception so that the client can understand the nature of the problem

Reply:
This exception is already in the application :) But agreed it should be a darts exception.
Reason for chucking the exception instead of just getting the first is to protect the data.

But I think this needs to be redesign following some internal chats we had as, we need to handle many to many relationships between hearing and events now as there are 2,000,000 records in production with this relation